sources/tools/Stride.Importer.3D/Material/Materials.cs

```diff
@@ -0,0 +1,152 @@
+        /// </summary>
+        public static readonly StackElementType[] ConvertAssimpStackTypeCppToCs = new StackElementType[]
+        {
+            StackElementType.Color,			// aiStackType_ColorType
+            StackElementType.Texture,		// aiStackType_TextureType
+            StackElementType.Operation      // aiStackType_BlemdOpType
+        };
+
+        public static readonly MappingMode[] ConvertAssimpMappingModeCppToCs = new MappingMode[]
+        {
+            MappingMode.Wrap,		// aiTextureMapMode_Wrap
+            MappingMode.Clamp,		// aiTextureMapMode_Clamp
+            MappingMode.Mirror,		// aiTextureMapMode_Mirror
+            MappingMode.Decal        // aiTextureMapMode_Decal
+        };
+
+        public static unsafe MaterialStack ConvertAssimpStackCppToCs(Silk.NET.Assimp.Assimp assimp, Silk.NET.Assimp.Material* material, Silk.NET.Assimp.TextureType type)
+        {
+            var ret = new MaterialStack();
+            var count = (int)assimp.GetMaterialTextureCount(material, type);
+
+            // Process the material stack
+            for (int iEl = count - 1; iEl >= 0; --iEl)
+            {
+                StackElement el;
+                // Common properties
+                int elType = 0, elFlags = 0;
+                float elAlpha = 0.0f, elBlend = 0.0f;
+                // Operation-specific properties
+                int elOp = 0;
+                // Color-specific properties
+                var elColor = new System.Numerics.Vector4();
+                // Texture-specific properties
+                var elTexPath = new AssimpString();
+                int elTexChannel = 0, elMappingModeU = 0, elMappingModeV = 0;
+                uint pMax = 0;
+
+                if (assimp.GetMaterialFloatArray(material, MatKeyTexAlphaBase, (uint)type, (uint)iEl, ref elAlpha, ref pMax) != Return.Success)
+                    elAlpha = 1.0f; // default alpha
+                if (assimp.GetMaterialFloatArray(material, Silk.NET.Assimp.Assimp.MaterialTexblendBase, (uint)type, (uint)iEl, ref elBlend, ref pMax) != Return.Success)
+                    elBlend = 1.0f; // default blend
+                if (assimp.GetMaterialIntegerArray(material, Silk.NET.Assimp.Assimp.MaterialTexflagsBase, (uint)type, (uint)iEl, ref elFlags, ref pMax) != Return.Success)
+                    elFlags = 0; // default flags (no flags)
+                if (assimp.GetMaterialIntegerArray(material, MatKeyTexTypeBase, (uint)type, (uint)iEl, ref elType, ref pMax) != Return.Success)
+                    elType = (int)StackElementType.Texture;//continue; // error !
+
+                switch ((StackElementType)elType)
+                {
+                    case StackElementType.Operation:
+                        if (assimp.GetMaterialIntegerArray(material, Silk.NET.Assimp.Assimp.MaterialTexopBase, (uint)type, (uint)iEl, ref elOp, ref pMax) != Return.Success)
+                            continue; // error !
```

Comment:
Best to output all cases of `// error !` to the asset importer's logger with an appropriate message if at all possible

Reply:
done, fixed!